src/picongpu/include/plugins/particleDetector/ParticleDetector.hpp

```diff
@@ -0,0 +1,537 @@
+        ParticlesType* particles = &(dc.getData<ParticlesType > (ParticlesType::FrameType::getName(), true));
+
+        ParticleDetectorKernel<typename ParticlesType::ParticlesBoxType,
+                               MyHistogramFunctor> particleDetectorKernel(
+            particles->getDeviceParticlesBox(),
+            *this->histogramFunctor);
+
+        /* create zone */
+        typedef typename MappingDesc::SuperCellSize SuperCellSize;
+
+        const PMacc::math::Int<simDim> coreBorderGuardSuperCells = this->cellDescription->getGridSuperCells();
+        const uint32_t guardSuperCells = this->cellDescription->getGuardingSuperCells();
+        const PMacc::math::Int<simDim> coreBorderSuperCells = coreBorderGuardSuperCells - 2*guardSuperCells;
+
+        /* this zone represents the core+border area with guard offset in unit of cells */
+        const zone::SphericZone<simDim> zone(
+            static_cast<PMacc::math::Size_t<simDim> >(coreBorderSuperCells * SuperCellSize::toRT()),
+            guardSuperCells * SuperCellSize::toRT());
+
+        /* kernel call */
+        algorithm::kernel::Foreach<SuperCellSize> foreach;
+        foreach(zone, cursor::make_MultiIndexCursor<simDim>(), particleDetectorKernel);
+
+        /* copy to host */
+        *this->hBufHistogram = *this->dBufHistogram;
+
+        /* mpi reduce */
+        using namespace lambda;
+        (*this->allGPU_reduce)(*this->hBufTotalHistogram, *this->hBufHistogram, _1 + _2);
+        if(!this->allGPU_reduce->root())
+            return;
+
+        this->writeToHDF5File(currentStep);
+    }
+
+
+    void setMappingDescription(MappingDesc* cellDescription)
+    {
+        this->cellDescription = cellDescription;
+    }
+
+
+    void pluginRegisterHelp(po::options_description& desc)
+    {
+        desc.add_options()
+        ((this->prefix + ".period").c_str(), po::value<uint32_t > (&this->notifyPeriod)->default_value(0),
+            "enable plugin [for each n-th step]")
+        ((this->prefix + ".numBinsTheta").c_str(), po::value<uint32_t > (&this->numBinsTheta)->default_value(64),
+            "number of bins for angle theta.")
+        ((this->prefix + ".numBinsPhi").c_str(), po::value<uint32_t > (&this->numBinsPhi)->default_value(64),
+            "number of bins for angle phi.")
+        ((this->prefix + ".numBinsEnergy").c_str(), po::value<uint32_t > (&this->numBinsEnergy)->default_value(1),
+            "number of bins for the energy spectrum. Disabled by default.")
+        ((this->prefix + ".minEnergy").c_str(), po::value<float_X > (&this->minEnergy)->default_value(float_X(0.0)),
+            "minimal detectable energy in keV.")
+        ((this->prefix + ".maxEnergy").c_str(), po::value<float_X > (&this->maxEnergy)->default_value(float_X(1.0e3)),
+            "maximal detectable energy in keV.")
+        ((this->prefix + ".logScale").c_str(), po::value<uint32_t > (&this->logScale)->default_value(0),
```

Comment:
here you emulate a bool flag with an integer. Please use a bool flag directly. An example of such a flag is [here](https://github.com/ComputationalRadiationPhysics/picongpu/blob/da64aace46f634cb4bd96a6ecdde020cd2046e16/src/picongpu/include/plugins/radiation/Radiation.hpp#L198)